### DISCUSSION TITLE
if else - how does it work for simple conditions?

### DISCUSSION THREAD

**Discussion by niloc7** [statamic/cms#4725] [Q&A] [answered]
I am trying to use the product-id from the parent page in the child pages.

Simple enough, all it needs is `{{ parent:product-id }}`

Okay, but there is also the main product page (which is the parent page) where I will simply use `{{ product-id }}` instead.

Question: How do I formulate the if else statement for this? :O (As they share the same template)

`{{ if parent:product-id }} {{ else product-id }}` doesn't seem to work.. well why would it.. doesn't seem logical :/

**Comment by niloc7** [comment, marked as answer]
This seems like a solution but it seems rather long.. is there a shorter way of doing this?

```
{{ if parent:product-id }}
                {{ parent:product-id }}
            {{ else }}
                {{ product-id }}
            {{ /if }}
```

**Reply by niloc7** [reply]
Ooh ternary operators!

`{{ product-id ? product-id : parent:product-id }}`

Hmm.. I wonder if there is an even more efficient way to do this.. or is this the best practice?

**Reply by jasonvarga** [reply]
`{{ product-id ?? parent:product-id }}`

https://statamic.dev/antlers#null-coalescence

**Reply by niloc7** [reply]
Wow that is a cool feature indeed!! Thanks Jason!
